File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/ManaBottle.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.entity.creature.player.Player;

public class ManaBottle extends Equipable {
	{
		name = Locale.get("mana_bottle");
		description = Locale.get("mana_bottle_desc");
		sprite = "item-mana_bottle";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).manaRegenRoom = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).manaRegenRoom = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/ManaBottle.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class ManaBottle : Equippable() {
	init {
		name = Locale.get("mana_bottle")
		description = Locale.get("mana_bottle_desc")
		sprite = "item-mana_bottle"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.manaRegenRoom = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.manaRegenRoom = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}